src/SDKs/Azure.Core/data-plane/Azure.Core.Tests/ETagTests.cs

```diff
@@ -0,0 +1,53 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.Collections.Generic;
+using System.Text;
+using NUnit;
+using NUnit.Framework;
+
+namespace Azure.Core.Tests
+{
+    public class ETagTests
+    {
+        [Test]
+        public void StringRoundtrips()
+        {
+            var s = "tag";
+            var eTag = new ETag(s);
+            Assert.AreSame(s, eTag.ToString());
+        }
+
+        [Test]
+        public void ComparisonOperators()
+        {
+            var eTag = new ETag("a");
+
```

Comment:
Unless I'm missing something, I believe that you missed the comparison of `ETag` to `object`, which is a dedicated public method in the implementation.

Reply:
Done